bin/dry-run.rb

```diff
@@ -294,16 +301,24 @@ def cached_dependency_files_read
   commit: $options[:commit]
 )
 
+$repo_contents_path = nil
+
+fetcher = Dependabot::FileFetchers.for_package_manager($package_manager).
+  new(source: source, credentials: $options[:credentials])
+
+if $options[:clone]
+  $repo_contents_path = fetcher.clone_repo_contents
```

Comment:
Yeah, good point, it would be great to add a file fetching strategy that fetches from disk, and the current API doesn't really allow for it. I can try to clean that up a bit even before we add support for the feature 👍

Reply:
I've spiked out some things and I think we should be able to do something like this, but since other systems currently depend on the given API, I'm going to leave this as is for now, once we start implementing fetching from disk, we can revisit?